src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+                "dist": None,
+                "platform": segment_span.get("platform"),  # column is always null in Clickhouse
+                "message": "",
+                "datetime": datetime.fromtimestamp(
+                    segment_span.get("precise.start_ts"), tz=timezone.utc
+                ).isoformat(),
+                "tags": list(segment_tags_items) + list(segment_sentry_tags.items()),
+                "_meta": {},
+                "_metrics": {},
+                "breakdowns": {
+                    "span_ops": _span_ops_breakdown(spans_data),
+                },
+                "contexts": {
+                    "browser": {
+                        "name": segment_sentry_tags.get("browser.name"),
+                        # other props missing from indexed spans dataset
+                    },
+                    "organization": {
+                        # missing from indexed spans dataset
+                    },
+                    "os": {
+                        "name": segment_sentry_tags.get("os.name"),
+                        # other props missing from indexed spans dataset
+                    },
+                    "trace": {
+                        "trace_id": segment_span.get("trace_id"),
+                        "span_id": segment_span.get("span_id"),
+                        "parent_span_id": parent_span_id,
+                        "op": segment_span.get("op"),
+                        "status": SPAN_STATUS_CODE_TO_NAME[segment_span.get("status", 2)],
+                        "exclusive_time": segment_span.get("exclusive_time"),
+                        "hash": segment_span.get("group"),
+                        "type": "trace",
+                    },
+                },
+                "culprit": transaction_name,
+                "environment": segment_sentry_tags.get("environment"),
+                "errors": [],
+                "extra": {},
+                "ingest_path": [],
+                "level": "info",
+                "location": transaction_name,
+                "logger": "",
+                "measurements": segment_measurements,
+                "metadata": {
+                    "location": transaction_name,
+                    "title": transaction_name,
+                },
+                "nodestore_insert": segment_span.get("precise.finish_ts"),
+                "received": segment_span.get("precise.finish_ts"),
+                "request": {
+                    # missing from indexed spans dataset
+                },
+                "sdk": {
+                    "name": segment_sentry_tags.get("sdk.name"),
+                    "version": segment_sentry_tags.get("sdk.version"),
+                    # other props missing from indexed spans dataset
+                },
+                "span_grouping_config": {
+                    # missing from indexed spans dataset
+                },
+                "spans": _span_data_to_event_spans(spans_data),
```

Comment:
Is this endpoint _just_ for trace view - since we're not planning to have an event details page? if so, I actually don't think we need to query spans at all - looking at trace view code - it seems like we're just using it for event metadata, since the trace endpoint returns all the spans. We can query the transaction dataset or just the segment span in this case?
If this is also meant to be a replacement for the `json/` endpoint, we should probably split it out into two so that we're not pulling spans when we don't need to for the trace view

Reply:
(for posterity, I already know you know this since you're the one who confirmed it 😅)
The trace view code uses the individual span data as well as the transaction metadata, so we do need to return all spans here.